frontend/model/contracts/identity.js

```diff
@@ -51,6 +60,26 @@ sbp('chelonia/defineContract', {
           Vue.set(state.settings, key, data[key])
         }
       }
+    },
+    'gi.contracts/identity/setLoginState': {
+      validate: objectOf({
+        groupIds: arrayOf(string)
+      }),
+      process ({ data }, { state }) {
+        Vue.set(state, 'loginState', data)
+      },
+      async sideEffect () {
+        try {
+          await sbp('gi.actions/identity/updateLoginStateUponLogin')
+        } catch (e) {
+          sbp('gi.notifications/emit', 'ERROR', {
+            message: L("Failed to groups we're part of with other computer. Not catastrophic, but could lead to problems. {errName}: '{errMsg}'", {
```

Comment:
Did you mean `"Failed to join groups... with another device..."`?

Reply:
Good catch, I see that it's not very clear, will reword